### PR TITLE
Issue# 307 (no confirmation)

### PR DESCRIPTION
**Pull request by Fan-Yang-284**
## Summary of Changes
Added transfer subcommand to coin. Includes checking for valid coin values (leq than user balance and geq 1) and the recipient is not the user. Currently does not include receiver confirmation, to be implemented.

## Related Issues
Issue #307

## Demonstration of Changes
image
image

## Further Information and Comments
Modified files: 
src/commands/coin.ts: added example coin transfer commands
.eslintrc: prettier formatting

New files:
src/commandDetails: transfer feature functionality

### CODE REVIEW

File: src/commandDetails/coin/transfer.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+const coinTransferExecuteCommand: SapphireMessageExecuteType = async (
+  client,
+  messageFromUser,
+  args,
+) => {
+  // First mandatory argument is receiving user
+  const receivingUser = <User>args['user'];
+  if (!receivingUser) {
+    throw new Error('please enter a valid user mention or ID for balance transfer.');
```
Comment (mcpenguin) [inline src/commandDetails/coin/transfer.ts:26]: Capitalize the error messages.

File: src/commandDetails/coin/transfer.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  }
+
+  // Second mandatory argument is amount
+  const amount = <number>args['amount'];
+  if (typeof amount !== 'number') {
```
Comment (mcpenguin) [inline src/commandDetails/coin/transfer.ts:31]: We should also check that amount is an integer. We don't want someone transferring 69.4204204204 coins and making the leaderboard ugly with the decimals.

**Review by mcpenguin — CHANGES_REQUESTED** [review]
Looks mostly good, left some comments.

File: src/commandDetails/coin/transfer.ts (hunk 1 above)
Comment (Fan-Yang-284) [reply src/commandDetails/coin/transfer.ts]: These should also be returning a SapphireMessageResponses and not throwing an error right - to notify the user?

**Review by mcpenguin — APPROVED** [review]
LGTM! Feel free to merge.

**Pull request merged**